src/hex.ts

```diff
@@ -0,0 +1,54 @@
+import { is, pattern, string } from 'superstruct';
+import { assert } from './assert';
+
+const HexStruct = pattern(string(), /^(?:0x)?[0-9a-f]+$/iu);
+
+/**
+ * Check if a string is a valid hex string.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is a valid hex string.
+ */
+export function isHexString(value: unknown): value is string {
+  return is(value, HexStruct);
+}
+
+/**
+ * Assert that a value is a valid hex string.
+ *
+ * @param value - The value to check.
+ * @throws If the value is not a valid hex string.
+ */
+export function assertIsHexString(value: unknown): asserts value is string {
+  assert(isHexString(value), 'Value must be a hexadecimal string.');
```

Comment:
I wonder if manual loop instead of regexp would be faster.

Might not necessarily be so, because regexps get compiled down to machine code, while JS might be interpreted.

Reply:
Did some tests with this. RegEx is slower at `string.length <= 2`, but after that it's faster, and gets relatively faster the longer the string gets, it seems.